lib/src/front_end/piece_factory.dart

```diff
@@ -1399,21 +1416,65 @@ mixin PieceFactory {
         canBlockSplitRight: canBlockSplitRight));
   }
 
-  /// Writes a [Piece] for the `<variable> in <expression>` part of a for-in
-  /// loop.
-  void writeForIn(AstNode leftHandSide, Token inKeyword, Expression sequence) {
+  /// Writes the `<variable> in <expression>` part of a for-in loop.
+  void _writeForIn(AstNode leftHandSide, Token inKeyword, Expression sequence) {
```

Comment:
[nit] With one method `_writeDeclaredForIn` having more specificity in the name than `_writeForIn` I anticipated some layering or difference in specificity that doesn't exist in the implementation. WDYT about something like  `_writeIdentifierForIn` ?